server/controllers/notification.js

```diff
@@ -2,69 +2,82 @@ const Notification = require("../models/Notification");
 const User = require("../models/User");
 const asyncHandler = require("express-async-handler");
 
-// @route PUT /profile/edit
-// @desc edit user profile
+// @route POST /notification/
+// @desc create a new notification
 // @access Public
 
 exports.createNotification = asyncHandler(async (req, res, next) => {
-const userId = await User.findById(req.user.id);
-const { type, title, description } = req.body;
-const notification = await Notification.create({userId, type, title, description});
+  const user = await User.findById(req.user.id);
+  const { type, title, description } = req.body;
+  const notification = await Notification.create({
+    user,
+    type,
+    title,
+    description,
+  });
 
   res.status(200).json({
     success: {
-      notification: notification
+      notification: notification,
     },
   });
-
 });
 
+// @route PUT /notification/mark-read/:notificationId
+// @desc changes mark as read property to true on a notification
+// @access Private
 
 exports.markAsRead = asyncHandler(async (req, res, next) => {
-const {notificationId} = req.params;
-const update = { read: true };
-const foundNotification = await Notification.findOneAndUpdate(notificationId, update);
-const readNotification = await foundNotification.save()
+  const { notificationId } = req.params;
+  const foundNotification = await Notification.findById(notificationId);
+
+  if (foundNotification.user.toString() !== req.user.id) {
+    res.status(403);
+    throw new Error("Forbidden");
+  }
```

Comment:
I changed this to res.status 403, should I throw an Error like this here?

Reply:
That should work, yes! I'd recommend sending a more descriptive error message though - `403` is named `Forbidden` so it's enough to convey that. You can send something like "The user doesn't have the correct privileges or permissions to modify this notification"👍🏼